src/block/compress.rs

```diff
@@ -324,117 +327,177 @@ pub fn backtrack_match(candidate: &mut usize, cur: &mut usize, literal_start: us
 ///
 #[inline]
 #[cfg(not(feature = "safe-encode"))]
-pub fn backtrack_match(candidate: &mut usize, cur: &mut usize, literal_start: usize, input: &[u8]) {
+fn backtrack_match(
+    input: &[u8],
+    cur: &mut usize,
+    literal_start: usize,
+    source: &[u8],
+    candidate: &mut usize,
+) {
     while unsafe {
         *candidate > 0
             && *cur > literal_start
-            && input.get_unchecked(*cur - 1) == input.get_unchecked(*candidate - 1)
+            && input.get_unchecked(*cur - 1) == source.get_unchecked(*candidate - 1)
     } {
         *cur -= 1;
         *candidate -= 1;
     }
 }
 
-/// Compress all bytes of `input` into `output`.
+/// Compress all bytes of `input[input_pos..]` into `output`.
+/// Bytes in `input[..input_pos]` are treated as a preamble and can be used for lookback.
+/// Bytes in `ext_dict` logically precede `pre[..input_pos]` and can also be used for lookback.
+/// `input_stream_offset` is the logical position of the first byte of `input` in the complete
```

Comment:
having a block here explaining the difference between input_pos and input_stream_offset would be good

Reply:
I added a few lines, but TBH it is quite hard to explain.